app/src/main/java/ltd/evilcorp/atox/activity/ChatActivity.kt

```diff
@@ -17,15 +18,25 @@ class ChatActivity : AppCompatActivity() {
         super.onCreate(savedInstanceState)
         setContentView(R.layout.activity_chat)
 
-        messages.add(MessageModel("hello", Sender.Received))
-        messages.add(MessageModel("how are you", Sender.Received))
-        messages.add(MessageModel("I'm good, thanks.", Sender.Sent))
-
         val adapter = MessagesAdapter(this, messages)
         val messageView = findViewById<ListView>(R.id.messages)
         messageView.adapter = adapter
 
+        val handler = Handler(Looper.getMainLooper()) {
+            true
+        }
+
         send.setOnClickListener {
+            val fnum: Int = intent.getIntExtra("fnum", 0)
+
+            val sendMsg = Message()
+            sendMsg.what = ToxThread.msgSendMsg
+            sendMsg.obj = outgoingMessage.text.toString()
+            sendMsg.arg1 = fnum
+
+            App.toxThread.handler.sendMessage(sendMsg)
+            App.toxThread.handler.sendEmptyMessage(ToxThread.msgSave)
```

Comment:
You don't need to save after sending a message.

Reply:
lol i dont know why i did that, done